pygmt/src/grdhisteq.py

```diff
@@ -321,23 +321,11 @@ def compute_bins(
         This method does a weighted histogram equalization for geographic
         grids to account for node area varying with latitude.
         """
-        # Return a pandas.DataFrame if ``outfile`` is not set
-        if output_type not in ["numpy", "pandas", "file"]:
-            raise GMTInvalidInput(
-                "Must specify 'output_type' either as 'numpy', 'pandas' or 'file'."
-            )
+        output_type = validate_output_type(output_type, outfile=outfile)
 
         if header is not None and output_type != "file":
             raise GMTInvalidInput("'header' is only allowed with output_type='file'.")
```

Comment:
Thinking if the order should be reversed here?
```suggestion
        if header is not None and output_type != "file":
            raise GMTInvalidInput("'header' is only allowed with output_type='file'.")

        output_type = validate_output_type(output_type, outfile=outfile)
```